include/ensmallen_bits/cmaes/cmaes_impl.hpp

```diff
@@ -313,6 +325,90 @@ typename MatType::elem_type CMAES<SelectionPolicyType>::Optimize(
   return overallObjective;
 }
 
+// Transforms the candidate into the given bounds.
+template<typename SelectionPolicyType>
+template<typename BaseMatType>
+void CMAES<SelectionPolicyType>::BoundaryTransform(BaseMatType& matrix)
+{
+  typedef typename BaseMatType::elem_type ElemType;
+  const double diff = (upperBound - lowerBound) / 2.0;
+  const double al = std::min(diff, (1 + std::abs(lowerBound)) / 20.0);
+  const double au = std::min(diff, (1 + std::abs(upperBound)) / 20.0);
+  const double xlow = lowerBound - 2 * al - diff;
+  const double xup = upperBound + 2 * au + diff;
+  const double r = 2 * (upperBound - lowerBound + al + au);
+
+  for (size_t col = 0; col < matrix.n_cols; col++)
+  {
+    for (size_t row = 0; row < matrix.n_rows; row++)
+    {
+      ElemType y = matrix(row, col);
+      // Boundary transformation shift into feasible pre-image.
+      if (y < xlow)
+      {
+        y += (ElemType)(r * (1 + (xlow - y) / r));
```

Comment:
Shouldn't this be` y += (ElemType)(r * (1 + (int)(xlow - y) / r));` also is the cast to `ElemType` needed, not sure I can see the reason to do that?

Reply:
Hi, @zoq sorry for replying so late. I did it to ensure that every element of the matrix are of same types if not required I'll remove it. Thanks.

Reply:
If you don't mind let's remove the extra cast.